### DISCUSSION TITLE
Be able to Zoom media

### DISCUSSION THREAD

**Discussion by Hi-there-how-are-u** [sircharlo/meeting-media-manager#413] [Ideas] [answered]
Would it be possible to add function to zoom media such as images? Sometimes we would like to zoom in, especially if there are text in pictures or if we would like to highlight part of the image.

**Comment by Chipperchap** [comment]
I would be interested in this too... 😁

**Comment by mtdvlpr** [comment]
I can see how that would be useful. I can definitely look into it. I'm just afraid that the Zoom would be zooming in the middle and not being able to Zoom into a particular part of the image (like the top right section). Would that still be something you would like to see?

**Reply by mtdvlpr** [reply]
Update: I found a package that allows you to zoom in and drag the image around, so it is definitely possible :)

**Comment by mtdvlpr** [comment]
@sircharlo, I made a working version for this in the branch `feat/zoom-pictures`. What do you think?

**Reply by sircharlo** [reply]
Nice idea, but two things I'm not sure of.
1- It would be nice if the pan and zoom could be achieved by hovering and interacting with the image preview in the controller window, not the image itself. To prevent users seeing the cursor.
2- How are users supposed to move the media window now?

**Reply by mtdvlpr** [reply]
1. Definitely! Not sure how easy it is to implement though
2. The image is only pannable when zoomed in, so when not zooming, you can just drag the window

**Reply by sircharlo** [reply]
Regarding 2, on Windows, the media window is no longer movable whether media is being displayed or not, due to the CSS changes you made...

**Reply by mtdvlpr** [reply]
You're right. My mistake. Fixed in https://github.com/sircharlo/meeting-media-manager/pull/592

**Reply by mtdvlpr** [reply]
For point 1. I managed to allow zoom in and out by hovering over the preview image in the controller. i am unsure how to implement the panning in a clean way. Actually dragging the preview image is a no go, so it has to be with either up/down/left/right buttons or keyboard shortcuts.

**Comment by mtdvlpr** [comment, marked as answer]
The feature has been merged and will be in the next release :). It's a bit of a discover yourself feature as there is no clear explanation. When you hover over the image preview. When the image is active, you can use your mouse wheel to zoom in and then drag the image to highlight the part of the image you want. The preview image will show the same view as the media window.

**Reply by joaochris** [reply]
> The feature has been merged and will be in the next release :). It's a bit of a discover yourself feature as there is no clear explanation. When you hover over the image preview. When the image is active, you can use your mouse wheel to zoom in and then drag the image to highlight the part of the image you want. The preview image will show the same view as the media window.

I couldn't get it to work. In fact, my mouse wheel broke 😂

It would be interesting for you to activate this feature in another way, in cases where we do not have the mouse wheel.

**Reply by mtdvlpr** [reply]
Hmm. Well, when you're on a laptop, you can use two fingers on the mouse pad to zoom in/out. I guess we could also implement a double click on the preview as a zoom action and when you double click and it's already maximally zoomed you zoom out again. @sircharlo, thoughts?

**Reply by sircharlo** [reply]
Double-click would toggle between 1x, 2x, 3x and 4x. It's not super elegant, but I suppose it would be a good alternative method.
